kernel/arch/i386/boot/boothelp.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file boothelp.cpp
+ * @author James T. Sprinkle (the-grue@hotmail.com)
+ * @brief
+ * @version 0.1
+ * @date 2021-08-07
+ *
+ * @copyright Copyright the Panix Contributors (c) 2021
+ *
+ */
+#include <stdint.h>
+
+#include <multiboot/multiboot2.h>
+#include <stivale/stivale2.h>
+
+extern "C" uint32_t stivale2_mmap_helper(void* baseaddr);
+extern "C" uint32_t multiboot2_mmap_helper(void* baseaddr);
+
+/**
+ *  ___ _   _          _     ___
+ * / __| |_(_)_ ____ _| |___|_  )
+ * \__ \  _| \ V / _` | / -_)/ /
+ * |___/\__|_|\_/\__,_|_\___/___|
+ *
+ * Scan the stivale2 tags to find the reclaimable bootloader
+ * memory map tag that starts at the stivale2 tag base
+ * address.  Return the length found.  We'll memory map
+ * the entire area.
+ *
+ */
+extern "C" uint32_t
+__attribute__ ((section(".early_text")))
+__attribute__((optimize("O0")))
+stivale2_mmap_helper(void* baseaddr)
+{
+    struct stivale2_tag* tag = (struct stivale2_tag*)baseaddr;
+
+    while (tag)
+    {
+        switch(tag->identifier)
+        {
+            case STIVALE2_STRUCT_TAG_MEMMAP_ID:
+            {
+                auto memmap = (struct stivale2_struct_tag_memmap*)tag;
+                for(uint32_t i = 0; i < (uint32_t)memmap->entries; i++)
+                {
+                    switch((uint32_t) memmap->memmap[i].type)
+                    {
+                        case STIVALE2_MMAP_BOOTLOADER_RECLAIMABLE:
```

Comment:
I think this code doesn't quite match the WebKit style guide we follow. You should be able to run it through `clang-format`, but I can also do that for you before we merge if you'd like, since I already have that setup.